plugins/big-number/src/BigNumber/transformProps.ts

```diff
@@ -19,25 +19,56 @@
 import * as color from 'd3-color';
 import { getNumberFormatter, NumberFormats } from '@superset-ui/number-format';
 import { ChartProps } from '@superset-ui/chart';
-import getTimeFormatterForGranularity from '../utils/getTimeFormatterForGranularity';
+import getTimeFormatterForGranularity, {
+  TimeGranularity,
+} from '../utils/getTimeFormatterForGranularity';
 
 const TIME_COLUMN = '__timestamp';
 const formatPercentChange = getNumberFormatter(NumberFormats.PERCENT_SIGNED_1_POINT);
 
+export interface DatasourceMetric {
+  label: string;
+  // eslint-disable-next-line camelcase
+  metric_name?: string;
+  d3format?: string;
+}
+
 // we trust both the x (time) and y (big number) to be numeric
-type BigNumberDatum = {
-  [TIME_COLUMN]: number;
+export interface BigNumberDatum {
   [key: string]: number | null;
+}
+
+export type BigNumberFormData = {
+  colorPicker?: {
+    r: number;
+    g: number;
+    b: number;
+  };
+  metric?:
+    | {
+        label: string;
+      }
+    | string;
+  compareLag?: string | number;
+  yAxisFormat?: string;
+  timeGrainSqla?: TimeGranularity;
+} & ChartProps['formData'];
+
+export type BignumberChartProps = ChartProps & {
+  formData: BigNumberFormData;
+  queryData: {
+    data?: BigNumberDatum[];
+  } & ChartProps['queryData'];
 };
 
-export default function transformProps(chartProps: ChartProps) {
-  const { width, height, formData, queryData } = chartProps;
+export default function transformProps(chartProps: BignumberChartProps) {
+  const { width, height, queryData, formData } = chartProps;
   const {
     colorPicker,
-    compareLag: compareLagInput,
+    compareLag: compareLag_,
```

Comment:
readability nit – I think `compareLagInput` is a clearer name, not sure what `_` means tho I guess it's used a few places here

Reply:
I use dangling "_" in for arguments that will be immediately overridden by simple argument processing. It indicates a sense of temporariness and keeps the variable names clean.